packages/rest/src/lib/handlers/BurstHandler.ts

```diff
@@ -0,0 +1,184 @@
+		const timeout = setTimeout(() => controller.abort(), this.manager.options.timeout).unref();
+		if (requestData.signal) {
+			// The type polyfill is required because Node.js's types are incomplete.
+			const signal = requestData.signal as PolyFillAbortSignal;
+			// If the user signal was aborted, abort the controller, else abort the local signal.
+			// The reason why we don't re-use the user's signal, is because users may use the same signal for multiple
+			// requests, and we do not want to cause unexpected side-effects.
+			if (signal.aborted) controller.abort();
+			else signal.addEventListener('abort', () => controller.abort());
+		}
+
+		let res: Dispatcher.ResponseData;
+		try {
+			res = await request(url, { ...options, signal: controller.signal });
+		} catch (error: unknown) {
+			if (!(error instanceof Error)) throw error;
+			// Retry the specified number of times if needed
+			if (shouldRetry(error) && retries !== this.manager.options.retries) {
+				// eslint-disable-next-line no-param-reassign
+				return await this.runRequest(routeId, url, options, requestData, ++retries);
+			}
+
+			throw error;
+		} finally {
+			clearTimeout(timeout);
+		}
+
+		const status = res.statusCode;
+		let retryAfter = 0;
+		const retry = parseHeader(res.headers['retry-after']);
+
+		// Amount of time in milliseconds until we should retry if rate limited (globally or otherwise)
+		if (retry) retryAfter = Number(retry) * 1_000 + this.manager.options.offset;
+
+		// Count the invalid requests
+		if (status === 401 || status === 403 || status === 429) {
+			this.manager.incrementInvalidCount();
+		}
+
+		if (status >= 200 && status < 300) {
+			return res;
+		} else if (status === 429) {
+			// Unexpected ratelimit
+			const isGlobal = res.headers['x-ratelimit-global'] !== undefined;
+			await onRateLimit(this.manager, {
+				timeToReset: retryAfter,
+				limit: Number.POSITIVE_INFINITY,
+				method,
+				hash: this.hash,
+				url,
+				route: routeId.bucketRoute,
+				majorParameter: this.majorParameter,
+				global: isGlobal,
+			});
+			this.debug(
+				[
+					'Encountered unexpected 429 rate limit',
+					`  Global         : ${isGlobal}`,
+					`  Method         : ${method}`,
+					`  URL            : ${url}`,
+					`  Bucket         : ${routeId.bucketRoute}`,
+					`  Major parameter: ${routeId.majorParameter}`,
+					`  Hash           : ${this.hash}`,
+					`  Limit          : ${Number.POSITIVE_INFINITY}`,
+					`  Retry After    : ${retryAfter}ms`,
+					`  Sublimit       : None`,
+				].join('\n'),
+			);
+
+			// We are bypassing all other limits, but an encountered limit should be respected (it's probably a non-punished ratelimit anyways)
+			await sleep(retryAfter);
+
+			// Since this is not a server side issue, the next request should pass, so we don't bump the retries counter
+			return this.runRequest(routeId, url, options, requestData, retries);
+		} else if (status >= 500 && status < 600) {
```

Comment:
While this is good to have, this is useless for interactions as if it times out, we're most likely past the 3s limit too. Leaving it up to you if we should keep this

Reply:
I left this in for the case where interaction callbacks aren't the only thing using this handler. That may never happen but if it does we let discord error which I think is better?